ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -247,12 +260,60 @@ NetworkClient getNetworkClient() {
     return networkClient;
   }
 
+  /**
+   * Returns a concatenated file path
+   * @param remoteReplicaInfo Info about remote replica
+   * @param fileName Name of file to write text to
+   * @return Returns a concatenated file path
+   */
+  protected String getFilePath(RemoteReplicaInfo remoteReplicaInfo, String fileName) {
+    switch (remoteReplicaInfo.getReplicaId().getReplicaType()) {
+      case CLOUD_BACKED:
+        // example: partitionId/fileName
+        return String.join(File.separator, replicationConfig.backupCheckerReportDir,
+            Long.toString(remoteReplicaInfo.getReplicaId().getPartitionId().getId()),
+            fileName);
+      case DISK_BACKED:
+        // example: partitionId/hostname/fileName
+        return String.join(File.separator, replicationConfig.backupCheckerReportDir,
+            Long.toString(remoteReplicaInfo.getReplicaId().getPartitionId().getId()),
+            remoteReplicaInfo.getReplicaId().getDataNodeId().getHostname(),
+            fileName);
+      default:
+        throw new IllegalArgumentException("Invalid replica type " + remoteReplicaInfo.getReplicaId().getReplicaType());
+    }
+  }
+
   /**
    * Logs replication progress of local node against some remote node
    * @param remoteReplicaInfo remote replica information
+   * @param exchangeMetadataResponse metadata information from remote node
    */
   protected void logReplicationStatus(RemoteReplicaInfo remoteReplicaInfo,
       ExchangeMetadataResponse exchangeMetadataResponse) {
+    if (replicationConfig.backupCheckFileManagerType.equals(replicationConfig.DEFAULT_BACKUP_CHECKER_FILE_MANAGER)) {
+      // optimization to not affect replication in servers in common case
```

Comment:
@JingQianCloud @ankagrawal Please take look here. This returns empty. So nothing is logged in common case.